web3_proxy/src/admin_queries.rs

```diff
@@ -0,0 +1,134 @@
+use crate::app::Web3ProxyApp;
+use crate::frontend::errors::FrontendErrorResponse;
+use crate::user_queries::get_user_id_from_params;
+use anyhow::Context;
+use axum::{
+    Json,
+    headers::{authorization::Bearer, Authorization},
+    TypedHeader,
+};
+use axum::response::{IntoResponse, Response};
+use entities::{admin, user, user_tier};
+use ethers::prelude::Address;
+use hashbrown::HashMap;
+use http::StatusCode;
+use migration::sea_orm::{self, ActiveModelTrait, ColumnTrait, EntityTrait, IntoActiveModel, QueryFilter};
+use log::info;
+use redis_rate_limiter::redis::AsyncCommands;
+use crate::frontend::errors::FrontendErrorResponse::AccessDenied;
+
+// TODO: Add some logic to check if the operating user is an admin
+// If he is, return true
+// If he is not, return false
+// This function is used to give permission to certain users
+
+
+pub async fn query_admin_modify_usertier<'a>(
+    app: &'a Web3ProxyApp,
+    bearer: Option<TypedHeader<Authorization<Bearer>>>,
+    params: &'a HashMap<String, String>
+) -> Result<Response, FrontendErrorResponse> {
+
+    // Quickly return if any of the input tokens are bad
+    let user_address: Vec<u8> = params
+        .get("user_address")
+        .ok_or_else(||
+            FrontendErrorResponse::StatusCode(
```

Comment:
This works, but I made a `FrontendErrorResponse::BadRequest` to make this less verbose.